File: plugin/src/main/java/io/github/pshevche/GradleActPluginPlugin.java
package io.github.pshevche;

import org.gradle.api.Project;
import org.gradle.api.Plugin;

/**
 * A simple 'hello world' plugin.
 */
public class GradleActPluginPlugin implements Plugin<Project> {
    public void apply(Project project) {
        // Register a task
        project.getTasks().register("greeting", task -> {
            task.doLast(s -> System.out.println("Hello from plugin 'io.github.pshevche.greeting'"));
        });
    }
}

File: plugin/src/main/kotlin/io/github/pshevche/GradleActPluginPlugin.kt
package io.github.pshevche

import org.gradle.api.Project
import org.gradle.api.Plugin

class GradleActPluginPlugin: Plugin<Project> {
    override fun apply(project: Project) {
        project.tasks.register("greeting") { task ->
            task.doLast {
                println("Hello from plugin 'io.github.pshevche.greeting'")
            }
        }
    }
}
